src/MGmol.cc

```diff
@@ -834,23 +832,9 @@ void MGmol<OrbitalsType>::initNuc(Ions& ions)
 
     Potentials& pot = hamiltonian_->potential();
 
+    // initialize poentials based on ionic positions and their species
     pot.initialize(ions);
 
-    // Check compensating charges
-    double comp_rho = getCharge(pot.rho_comp());
-
-    if (onpe0 && ct.verbose > 1)
-    {
-        os_ << std::setprecision(8) << std::fixed
-            << " Charge of rhoc: " << comp_rho << std::endl;
-    }
-
-#if 1
-    pot.rescaleRhoComp();
-#endif
-
-    pot.addBackgroundToRhoComp();
```

Comment:
@jeanlucf22, We don't need the background charge and rescaling of rho anymore? Was this a bug before?

Reply:
It is all included in the initialize() function now

Reply:
Got it. I missed the expansion earlier.